lib/source/consul.js

```diff
@@ -0,0 +1,292 @@
+/* global Log, Config */
+'use strict';
+
+const EventEmitter = require('events').EventEmitter;
+const consul = require('consul');
+
+const DEFAULT_AGENT_ADDRESS = '127.0.0.1';
+const DEFAULT_AGENT_PORT = 8500;
+
+class Consul extends EventEmitter {
+  /**
+   * Creates a new instance of a Consul souce plugin
+   *
+   * The following options are valid:
+   * - host (String, default 127.0.0.1): Consul agent's address
+   * - port (Number, default 8500) Consul agent's HTTP(S) port
+   * - secure (Boolean, default: true): enable HTTPS
+   *
+   * @param {Object} options  Options that can be set for the plugin
+   */
+  constructor(options) {
+    super();
+
+    const configuration = options || Object.create(null);
+
+    this.type = 'consul';
+    this.name = 'consul';
+    this.properties = Object.create(null);
+    this.host = configuration.host || DEFAULT_AGENT_ADDRESS;
+    this.port = configuration.port || DEFAULT_AGENT_PORT;
+    this.secure = configuration.secure !== false;
+    this._okay = false;
+  }
+
+  /**
+   * Start watching the Consul agent for changes
+   */
+  initialize() {
+    if (this._consul) {
+      return;
+    }
+
+    this._consul = consul({
+      host: this.host,
+      port: this.port,
+      secure: this.secure
+    });
+
+    this._serviceWatcher = this._consul.watch({
+      method: this._consul.catalog.service.list
+    });
+
+    this._serviceWatcher.on('error', (error) => {
+      this._error(error);
+    });
+
+    this._serviceWatcher.on('change', (serviceListData) => {
+      this._onServiceListChange(serviceListData);
+    });
+
+    this._okay = true;
+    this.emit('startup');
+  }
+
+  /**
+   * Stop watching the Consul agent for changes
+   */
+  shutdown() {
+    this._shutdownHealthWatchers();
+    this._shutdownServiceWatcher();
+    delete this._consul;
+
+    this._okay = false;
+    this.emit('shutdown');
+  }
+
+  /**
+   * Return an object describing the Consul plugin's current status
+   *
+   * @return {Object}
+   */
+  status() {
+    return {
+      okay: this._okay,
```

Comment:
To be consistent with the rest of the `Source#status()` output, this should be `ok` rather than `okay`.